esp/services/ws_dfu/ws_dfuService.cpp

```diff
@@ -6118,7 +6104,289 @@ bool CWsDfuEx::onDFUFileAccess(IEspContext &context, IEspDFUFileAccessRequest &r
             userDesc.setown(createUserDescriptor());
             userDesc->set(userID.str(), context.queryPassword(), context.querySignature());
         }
-        getFileAccess(context, userDesc, accessType, req, resp);
+        getFileAccess(context, userDesc, accessType, requests.getName(), requests.getJobId(), requests, resp.updateAccessInfo());
+    }
+    catch (IException *e)
+    {
+        FORWARDEXCEPTION(context, e,  ECLWATCH_INTERNAL_ERROR);
+    }
+    return true;
+}
+
+const char *CWsDfuEx::createWsDFUCreateGroupName(StringArray &locations, StringBuffer &groupName)
+{
+    if (locations.length() < 1)
+        return nullptr;
+
+    StringBuffer buf;
+    ForEachItemIn(i, locations)
+        buf.append(locations.item(i)).append(";");
+    groupName.set("wsdfucreate").append(hashc((unsigned char*) buf.str(), buf.length(), 0));
```

Comment:
is a 32bit hash good enough, probably, but do you check that if exists (published group) it's actually the same list of locations ?

Reply:
The createDFUDescriptor() does check the groupName. If the groupName exists, I use the old one. Is it good enough?